cql/DashboardLibrary.cql

```diff
@@ -786,13 +788,47 @@ define function CollateConclusionCodes(D DiagnosticReport, ObsList List<Observat
   else
     ValuesFromObservation(D, ObsList)
 
+define function ToFhirCodeableConcept(c System.Code):
+    FHIR.CodeableConcept {
+      coding: {
+        FHIR.Coding {
+          system: FHIR.uri {value: c.system},
+          code: FHIR.code {value: c.code},
+          display: FHIR.string {value: c.display}
+        }
+      }
+    }
+
+define Hpv16CodeToFhirCodeableConcept:
+  ToFhirCodeableConcept("HPV16+")
+
+define Hpv18CodeToFHIRCodeableConcept:
+  ToFhirCodeableConcept("HPV18+")
+
+// If an HPV test has a code for a genotyped high-risk HPV test, and a generic value representing a positive test,
+// this function returns the valueCodeableConcept that would represent the positive genotyped high-risk test,
+// as opposed to the generic positive value
+// @param O - a FHIR Observation
+// @returns {FHIR.CodeableConcept} the valueCodeableConcept of the Observation,
+// or the CodeableConcept of HPV16+ or HPV18+ if the test was positive for code of HPV16 or HPV18
+define function ConvertHpvTestTypeToValue(O Observation):
+  if O.value as FHIR.CodeableConcept in "High Risk HPV Positive Results" then
+    if O.code in "HPV 16 Tests" then 
+      Hpv16CodeToFhirCodeableConcept
+    else if O.code in "HPV 18 Tests" then 
+      Hpv18CodeToFHIRCodeableConcept
+    else
+      O.value 
```

Comment:
convert to CodeableConcept to match the return type.

Reply:
Also done: https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/117/commits/938d71d318b5a23471dc3c0d8a7c1b81b3eebe2e.